tools/clang/include/clang/SPIRV/SpirvVisitor.h

```diff
@@ -114,12 +115,16 @@ class Visitor {
 #undef DEFINE_VISIT_METHOD
 
 protected:
-  explicit Visitor(const SpirvCodeGenOptions &opts) : spvOptions(opts) {}
+  explicit Visitor(const SpirvCodeGenOptions &opts, SpirvContext &ctx,
+                   SpirvModule &mod)
+      : spvOptions(opts), context(ctx), module(mod) {}
 
   const SpirvCodeGenOptions &getCodeGenOptions() const { return spvOptions; }
 
-private:
+protected:
   const SpirvCodeGenOptions &spvOptions;
+  SpirvContext &context;
+  SpirvModule &module;
```

Comment:
Do we need to keep a reference to the module?

The intended usage of the visitor is to override suitable methods to only handle the entities of interest, like instructions or basic blocks. So it's up to the concrete visitor to decide which entity is of interest. If they want the module, they can grab it at the time of `visit(Module)`.